js/components/thumbnail.js

```diff
@@ -7,12 +7,107 @@ var React = require('react'),
+    var pitch = this.props.vrViewingDirection.pitch;
+    this.setCurrentViewVr(yaw, pitch);
+  },
+  componentWillReceiveProps: function(nextProps) {
+    if (this.props.vrViewingDirection !== nextProps.vrViewingDirection) {
+      var yaw = nextProps.vrViewingDirection.yaw;
+      var pitch = nextProps.vrViewingDirection.pitch;
+      this.setCurrentViewVr(yaw, pitch);
+    }
+  },
   shouldComponentUpdate: function(nextProps) {
-    return (nextProps.hoverPosition != this.props.hoverPosition);
+    var hoverPosition = nextProps.hoverPosition != this.props.hoverPosition;
+    var fullscreen  = nextProps.fullscreen != this.props.fullscreen && this.props.videoVr;
+    var vrViewingDirection = nextProps.vrViewingDirection != this.props.vrViewingDirection;
+    return (hoverPosition || fullscreen || vrViewingDirection);
+  },
+  componentDidUpdate: function(prevProps, prevState) {
+    var newThumbnailWidth = $("#oo-thumbnail").width();
+    var newThumbnailHeight = $("#oo-thumbnail").height();
+    if (newThumbnailWidth !== this.thumbnailWidth || newThumbnailHeight !== this.thumbnailHeight) {
+      this.thumbnailWidth = newThumbnailWidth;
+      this.thumbnailHeight = newThumbnailHeight;
+      var yaw = this.props.vrViewingDirection.yaw;
+      var pitch = this.props.vrViewingDirection.pitch;
+      this.setCurrentViewVr(yaw, pitch);
+    }
+  },
+
+  setThumbnailSizes: function() {
+    var thumbnailWidth = $("#oo-thumbnail").width();
+    var thumbnailHeight = $("#oo-thumbnail").height();
+    if (thumbnailWidth) {
+      this.thumbnailWidth = thumbnailWidth;
+    }
+    if (thumbnailHeight) {
+      this.thumbnailHeight = thumbnailHeight;
+    }
+  },
+
+  setImageSizes: function() {
+    var thumbnail = Utils.findThumbnail(this.props.thumbnails, this.props.hoverTime, this.props.duration, this.props.videoVr);
+    this.imageWidth = thumbnail.imageWidth;
+    this.imageHeight = thumbnail.imageHeight;
+  },
+
+  /**
+   * @description set positions for a thumbnail image when a video is vr
+   * @param {Number} yaw - rotation around the vertical axis in degrees (returns after changing direction)
+   * @param {Number} pitch - rotation around the horizontal axis in degrees (returns after changing direction)
+   * @private
+   */
+  setCurrentViewVr: function(yaw, pitch) {
+    yaw = Utils.ensureNumber(yaw, 0);
+    pitch = Utils.ensureNumber(pitch, 0);
+    var imageWidth = this.imageWidth;
+    var imageHeight = this.imageHeight;
+    var thumbnailWidth = this.thumbnailWidth;
+    var thumbnailHeight = this.thumbnailHeight;
+    yaw = this.getCurrentYawVr(yaw);
+    pitch = pitch >= 360 ? 0 : pitch;
+
+    var positionY = -(((imageHeight - thumbnailHeight) / 2) - pitch);
+    var bottomCoordinate = -(imageHeight - thumbnailHeight);
+    if (positionY > 0) {
+      positionY = 0;
+    } else if (positionY < bottomCoordinate) {
+      positionY = bottomCoordinate;
+    }
+    var positionX = -(imageWidth - thumbnailWidth/2 - imageWidth*yaw/360);
```

Comment:
Can you fix the spacing on imageWidth*yaw/360 ?

Reply:
\+